src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1099,7 +1099,20 @@ private async Task<Stream> InternalGetRequestStream()
                 TaskCompletionSource<Stream> getStreamTcs = new();
                 TaskCompletionSource completeTcs = new();
                 _sendRequestTask = SendRequest(async: true, new RequestStreamContent(getStreamTcs, completeTcs));
-                _requestStream = new RequestStream(await getStreamTcs.Task.ConfigureAwait(false), completeTcs);
+                Task<Stream> getStreamTask = getStreamTcs.Task;
+                try
+                {
+                    Task result = await Task.WhenAny((Task)getStreamTask, (Task)_sendRequestTask).WaitAsync(TimeSpan.FromMilliseconds(Timeout)).ConfigureAwait(false);
```

Comment:
The one thing that I've noticed is that, when user is using async path already they should implement their own timeout mechanism according to the doc: [HttpWebRequest.Timeout - Remarks](https://learn.microsoft.com/en-us/dotnet/api/system.net.httpwebrequest.timeout?view=net-9.0#remarks)

Not sure, if we should stick to that behavior and limit propagating of Timeout here.

Reply:
`WaitAsync` will not cancel / stop the underlying tasks. What happens with them? I think this will at minimum generate unobserved exceptions.
It would be better to pass the timeout to the tasks themselves via cancellation token and them catch the exception and remap to `TimeoutException` if necessary. (we do this a lot around `HttpClient`).